service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceTwinDevice.java

```diff
@@ -10,218 +10,169 @@
 import com.microsoft.azure.sdk.iot.deps.util.Tools;
 
-import java.util.HashSet;
-import java.util.Map;
-import java.util.Set;
+import java.util.*;
 
 /**
- * The DeviceTwinDevice class represent the twin on iotHub.
- * implementing constructors and serialization functionality.
- * If object is a representation of the module twin if and only if the moduleId is set.
+ * Represent the twin on IoT hub. Implementing constructors and serialization functionality.
+ * <p>The object is a representation of a module twin if and only if the moduleId is set.</p>
  */
 public class DeviceTwinDevice
 {
-    /**
-     *Codes_SRS_DEVICETWINDEVICE_25_001: [** The DeviceTwinDevice class has the following properties: deviceId, a container for tags, desired and reported properties, and a twin object. **]**
-     */
-    private String deviceId;
-    private String moduleId;
-    private String eTag;
+    private String deviceId = null;
+    private String moduleId = null;
+    private String eTag = null;
     private Integer version;
     private TwinCollection tag = null;
     private TwinCollection reportedProperties = null;
     private TwinCollection desiredProperties = null;
     private Map<String, ConfigurationInfo> configurations = null;
     private DeviceCapabilities capabilities = null;
-    private String connectionState;
-    private String modelId;
+    private String connectionState = null;
+    private String modelId = null;
+    private final String deviceScope = null;
+    private final List<String> parentScopes = new ArrayList<>();
 
     /**
-     * Constructor to create instance for a device
+     * Constructor to create an instance for a device.
      */
     public DeviceTwinDevice()
     {
-        /*
-        **Codes_SRS_DEVICETWINDEVICE_25_003: [** The constructor shall create a new instance of twin object for this device and store the device id.**]**
-         */
-        this.deviceId = null;
-        this.moduleId = null;
-        this.eTag = null;
-        this.version = null;
-        this.modelId = null;
     }
 
     /**
      * Constructor to create instance for a device.
      *
-     * @param deviceId Device ID for this device
-     * @throws IllegalArgumentException This exception is thrown if the device id is null or empty
+     * @param deviceId Id for this device.
+     * @throws IllegalArgumentException This exception is thrown if the device Id is {@code null} or empty
      */
     public DeviceTwinDevice(String deviceId) throws IllegalArgumentException
     {
         this();
 
         if (Tools.isNullOrEmpty(deviceId))
         {
-            /*
-            **Codes_SRS_DEVICETWINDEVICE_25_002: [** The constructor shall throw IllegalArgumentException if the input string is empty or null.**]**
-             */
-            throw new IllegalArgumentException("Device ID cannot be null or empty");
+            throw new IllegalArgumentException("Device Id cannot be null or empty.");
```

Comment:
I see some places you are using param name, to be consistent here it should be "deviceId cannot be ..."

Reply:
Updated